scipy/special/_basic.py

```diff
@@ -2352,14 +2358,14 @@ def factorial(n, exact=False):
                     current = un[i + 1]
                     val *= _range_prod(prev, current)
                     out[n == current] = val
+
+            if np.isnan(n).any():
```

Comment:
Maybe in a follow up store the `np.isnan` array so that it doesn't get computed so many times.